hub/core/chunk_engine/write.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import pickle
+from uuid import uuid1
+
+from hub.core.chunk_engine import generate_chunks
+
+from hub.core.typing import Provider
+from typing import Any, Callable, List, Tuple
+
+from .util import (
+    array_to_bytes,
+    normalize_and_batchify_shape,
+    get_meta_key,
+    get_index_map_key,
+    get_chunk_key,
+)
+
+
+def write_array(
```

Comment:
This one function is doing a lot of things. It would be better to break this down into smaller methods.